coordinator/txmanager.go

```diff
@@ -126,9 +129,37 @@ func (t *TxManager) syncSCVars(vars common.SCVariablesPtr) {
 
 // NewAuth generates a new auth object for an ethereum transaction
 func (t *TxManager) NewAuth(ctx context.Context, batchInfo *BatchInfo) (*bind.TransactOpts, error) {
-	gasPrice, err := t.ethClient.EthSuggestGasPrice(ctx)
-	if err != nil {
-		return nil, tracerr.Wrap(err)
+	//First we try getting the gas price from etherscan. If it fails we get the gas price from the ethereum node.
+	var gasPrice *big.Int
+	if t.etherscanService != nil {
+		etherscanGasPrice, err := (*t.etherscanService).GetGasPrice(ctx)
+		if err != nil {
+			log.Warn("Error getting the gas price from etherscan. Trying another method. Error: ", err.Error())
+			var er error
+			gasPrice, er = t.ethClient.EthSuggestGasPrice(ctx)
+			if er != nil {
+				return nil, tracerr.Wrap(er)
+			}
+		} else {
+			eGasPrice := new(big.Int)
+			eGasPrice, ok := eGasPrice.SetString(etherscanGasPrice.ProposeGasPrice, 10)
+			if !ok {
+				log.Warn("invalid big int: \"%v\"", etherscanGasPrice.ProposeGasPrice, ". Trying another method to get gas price")
+				var er error
+				gasPrice, er = t.ethClient.EthSuggestGasPrice(ctx)
+				if er != nil {
+					return nil, tracerr.Wrap(er)
+				}
+			} else {
+				gasPrice = eGasPrice
+			}
+		}
+	} else {
+		var er error
```

Comment:
Why do you need to create an additional variable **er**, and not use the previously declared variable **err**?